### PR TITLE
feat(IT Wallet): [SIW-623] Add credential issuance

### PR DESCRIPTION
**Pull request by LazyAfternoons**
## Short description
Co-authored by @balanza 

This PR adds the credential issuance flow to the app. 

## List of changes proposed in this pull request
**Implements actions, sagas and a reducer to handle the issuance flow.** 

  Preliminary Checks (checks pot):
  1. The `itwCredentialsCatalogScreen` performs preliminary checks when a credential is selected.
  2. The result of these checks is shown in `itwCredentialsCheckScreen`.
  3. The associated saga for this step is `itwIssuanceChecksSaga`.

  Authorization:
  1. `ItwCredentialAuthScreen` screen is currently mocked and displays a list of PID claims needed to obtain the credential.
  It doesn't need to be connected to a saga and doesn't dispatch any actions.

  Credential issuance (issuanceResult pot):
  1. Upon user confirmation, `ItwCredentialPreviewScreen` dispatches an action.
  2. The dispatched action triggers the `itwIssuanceGetCredentialSaga`, which obtains the credential and shows the result.

  Storing the Credential:
  1. If the user decides to store the credential, an action is dispatched and triggers the `addCredentialWithPin` which shows a PIN screen.
  2. It handles the PIN verification and eventually dispatches the necessary actions to store the credential.

## How to test
Activate the wall and then from the wallet section try to obtain an available credential.

### CODE REVIEW

**Review by hevelius — CHANGES_REQUESTED** [review]

File: ts/features/it-wallet/screens/credential/issuing/ItwCredentialChecksScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+      </SafeAreaView>
+    </BaseScreenComponent>
+  );
+
+  /**
+   * Error mapping function for any error that can be displayed in this screen.
+   * @param error - optional ItWalletError to be displayed.
+   * @returns a mapped error object or a generic error object when error is not provided.
+   */
+  const getScreenError: ItwErrorMapping = (error?: ItWalletError) => {
```
[the remark this answers is not in the document]
Comment (LazyAfternoons) [reply ts/features/it-wallet/screens/credential/issuing/ItwCredentialChecksScreen.tsx:100]: Yes, once we figure out if this is the way we want to manage them.

File: ts/features/it-wallet/saga/new/itwIssuanceSaga.ts [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+ * @returns the PID and the crypto context.
+ */
+function* getPID(): Iterator<any, readonly [string, CryptoContext]> {
+  const maybePid = yield* select(ItwCredentialsPidSelector);
+  if (O.isNone(maybePid)) {
+    const message = `Expecting response_code from sendAuthorizationResponse, received undefined`;
+    throw new Error(message);
+  }
+  const pidKeytag = ITW_PID_KEY_TAG;
+  const pidCryptoContext = createCryptoContextFor(pidKeytag);
```
[the remark this answers is not in the document]
Comment (LazyAfternoons) [reply ts/features/it-wallet/saga/new/itwIssuanceSaga.ts:386]: Done in 6d2c5ce thanks.

File: ts/features/it-wallet/saga/new/itwIssuanceSaga.ts [hunk 3]
```diff
@@ -0,0 +1,390 @@
...
+} from "../../../../store/actions/identification";
+import I18n from "../../../../i18n";
+import NavigationService from "../../../../navigation/NavigationService";
+import ROUTES from "../../../../navigation/routes";
+
+/**
+ * Watcher for issuance related sagas.
+ */
+export function* watchItwIssuanceSaga(): SagaIterator {
+  yield* takeLatest(itwIssuanceChecks.request, itwIssuanceChecksSaga);
```
[the remark this answers is not in the document]
Comment (LazyAfternoons) [reply ts/features/it-wallet/saga/new/itwIssuanceSaga.ts:48]: Yes, done in 3cfe29e. Thanks.

**Review by hevelius — APPROVED** [review]
LGTM. Tested on real device. I noticed that both in the preview and in the presentation test the credential detail shows the date in the YYYY-MM-DD format. ex (1970-04-22). Does it seem like it doesn't use the locale system configuration?

**Comment by LazyAfternoons** [comment]
> LGTM. Tested on real device. I noticed that both in the preview and in the presentation test the credential detail shows the date in the YYYY-MM-DD format. ex (1970-04-22). Does it seem like it doesn't use the locale system configuration?

That's true, it currently doesn't use the system locale. It requires a dedicated function which checks for each claim if it's a date or not and parse it accordingly. 
I create an issue (SIW-664) to track it.

**Pull request merged**